packages/world/ts/config/v2/worldWithShorthands.ts

```diff
@@ -62,17 +62,32 @@ export function resolveWorldWithShorthands<world extends WorldWithShorthandsInpu
   world: world,
 ): resolveWorldWithShorthands<world> {
   const scope = extendedScope(world);
-  const tables = mapObject(world.tables ?? {}, (table) => {
-    return isTableShorthandInput(table) ? resolveTableShorthand(table, scope) : table;
-  });
-  const namespaces = mapObject(world.namespaces ?? {}, (namespace) => ({
-    ...namespace,
-    tables: mapObject(namespace.tables ?? {}, (table) => {
-      return isTableShorthandInput(table) ? resolveTableShorthand(table, scope) : table;
-    }),
-  }));
 
-  const fullConfig = { ...world, tables, namespaces };
+  const fullConfig = {
+    ...world,
+    ...(world.tables
+      ? {
+          tables: mapObject(world.tables, (table) =>
+            isTableShorthandInput(table) ? resolveTableShorthand(table, scope) : table,
+          ),
+        }
+      : null),
+    ...(world.namespaces
```

Comment:
The previous `resolveWorldWithShorthands` code was always setting the `namespaces` key, which kept the downstream `resolveWorld` function from determining if we're using namespaces or not.